File: frontend/src/config/navigation.ts
/**
 * Configurația de navigare pentru site-ul primăriei
 * Structură organizată ierarhic cu icoane și categorii
 */
import React from 'react';
import {
  Business,
  Info,
  Gavel,
  Security,
  Article,
  RoomService,
  Payment,
  Groups,
  Campaign,
  Schedule,
  ContactSupport,
  Map,
  Assignment,
  Search
} from '@mui/icons-material';

export interface NavigationItem {
  label: string;
  path: string;
  icon?: React.ReactElement;
  children?: NavigationItem[];
  description?: string;
  category?: 'main' | 'service' | 'legal' | 'other';
}

// Structura de navigare îmbunătățită cu submeniuri
export const navigationItems: NavigationItem[] = [
  {
    label: 'Despre Primărie',
    path: '/despre-primarie',
    icon: React.createElement(Business),
    category: 'main',
    description: 'Informații despre organizarea și funcționarea primăriei',
    children: [
      { 
        label: 'Organizare', 
        path: '/despre-primarie/organizare', 
        icon: React.createElement(Business),
        description: 'Structura organizatorică și compartimente'
      },
      { 
        label: 'Conducere', 
        path: '/despre-primarie/conducere', 
        icon: React.createElement(Groups),
        description: 'Conducerea primăriei și consiliul local'
      },
      { 
        label: 'Strategia de Dezvoltare', 
        path: '/despre-primarie/strategia-dezvoltare', 
        icon: React.createElement(Map),
        description: 'Strategia de dezvoltare locală și proiecte'
      }
    ]
  },
  {
    label: 'Informații Publice',
    path: '/informatii-interes-public',
    icon: React.createElement(Info),
    category: 'main',
    description: 'Informații de interes public conform legii transparenței',
    children: [
      { 
        label: 'Buget și Execuție', 
        path: '/informatii-interes-public/buget', 
        icon: React.createElement(Payment),
        description: 'Bugetul local și execuția bugetară'
      },
      { 
        label: 'Achiziții Publice', 
        path: '/informatii-interes-public/achizitii', 
        icon: React.createElement(Assignment),
        description: 'Proceduri de achiziții publice'
      },
      { 
        label: 'Taxe și Impozite', 
        path: '/informatii-interes-public/taxe-impozite', 
        icon: React.createElement(Payment),
        description: 'Taxele și impozitele locale'
      }
    ]
  },
  {
    label: 'Transparență',
    path: '/transparenta-decizionala',
    icon: React.createElement(Gavel),
    category: 'main',
    description: 'Transparența procesului decizional',
    children: [
      { 
        label: 'Proiecte Hotărâri', 
        path: '/transparenta-decizionala/proiecte-hotarari', 
        icon: React.createElement(Article),
        description: 'Proiectele de hotărâri în dezbatere'
      },
      { 
        label: 'Ședințe Consiliul Local', 
        path: '/transparenta-decizionala/sedinte-consiliu', 
        icon: React.createElement(Schedule),
        description: 'Programul și procesele-verbale ale ședințelor'
      }
    ]
  },
  {
    label: 'Integritate',
    path: '/integritate-institutionala',
    icon: React.createElement(Security),
    category: 'main',
    description: 'Măsuri de integritate instituțională',
    children: [
      { 
        label: 'Cod Etic', 
        path: '/integritate-institutionala/cod-etic', 
        icon: React.createElement(Security),
        description: 'Codul etic și deontologic'
      },
      { 
        label: 'Plan Integritate', 
        path: '/integritate-institutionala/plan-integritate', 
        icon: React.createElement(Assignment),
        description: 'Planul de integritate instituțională'
      }
    ]
  },
  {
    label: 'Servicii Online',
    path: '/servicii-publice',
    icon: React.createElement(RoomService),
    category: 'service',
    description: 'Servicii publice digitale pentru cetățeni',
    children: [
      { 
        label: 'Programări Online', 
        path: '/programari-online', 
        icon: React.createElement(Schedule),
        description: 'Programează-te online la primărie'
      },
      { 
        label: 'Sesizări Cetățene', 
        path: '/servicii-publice/sesizari/formular', 
        icon: React.createElement(Campaign),
        description: 'Trimite o sesizare online'
      },
      { 
        label: 'Formulare Administrative', 
        path: '/servicii-publice/formulare', 
        icon: React.createElement(Assignment),
        description: 'Completează formulare online'
      },
      { 
        label: 'Verificare Sesizări', 
        path: '/servicii-publice/cautare-sesizare', 
        icon: React.createElement(Search),
        description: 'Verifică statusul unei sesizări'
      },
      { 
        label: 'Verificare Programări', 
        path: '/verificare-programare', 
        icon: React.createElement(Schedule),
        description: 'Verifică statusul unei programări'
      }
    ]
  },
  {
    label: 'Plăți Online',
    path: '/plati-online',
    icon: React.createElement(Payment),
    category: 'service',
    description: 'Plătește taxele și impozitele online',
    children: [
      { 
        label: 'Plata Taxelor', 
        path: '/plati-online', 
        icon: React.createElement(Payment),
        description: 'Plătește taxele locale online'
      },
      { 
        label: 'Verificare Plăți', 
        path: '/verificare-plati', 
        icon: React.createElement(Search),
        description: 'Verifică statusul unei plăți'
      }
    ]
  }
];

// Navigare suplimentară pentru footer și meniuri secundare
export const additionalNavItems: NavigationItem[] = [
  {
    label: 'Anunțuri',
    path: '/anunturi',
    icon: React.createElement(Campaign),
    category: 'main',
    description: 'Anunțuri oficiale ale primăriei'
  },
  {
    label: 'Monitorul Oficial Local',
    path: '/monitorul-oficial-local',
    icon: React.createElement(Article),
    category: 'main',
    description: 'Monitorul Oficial Local - MOL',
    children: [
      { label: 'Statutul UAT', path: '/mol/statutul-uat', icon: React.createElement(Article) },
      { label: 'Regulamente', path: '/mol/regulamente', icon: React.createElement(Article) },
      { label: 'Hotărâri', path: '/mol/hotarari', icon: React.createElement(Article) },
      { label: 'Dispoziții', path: '/mol/dispozitii', icon: React.createElement(Article) },
      { label: 'Documente Financiare', path: '/mol/documente-financiare', icon: React.createElement(Payment) },
      { label: 'Alte Documente', path: '/mol/alte-documente', icon: React.createElement(Article) }
    ]
  },
  {
    label: 'Comunitate',
    path: '/comunitate',
    icon: React.createElement(Groups),
    category: 'other',
    description: 'Informații despre comunitatea locală',
    children: [
      { label: 'Educație și Cultură', path: '/comunitate/educatie-cultura', icon: React.createElement(Groups) },
      { label: 'Mediu și Turism', path: '/comunitate/mediu-turism', icon: React.createElement(Map) }
    ]
  },
  {
    label: 'Contact',
    path: '/contact',
    icon: React.createElement(ContactSupport),
    category: 'other',
    description: 'Date de contact și program'
  }
];

// Linkuri legale pentru footer
export const legalNavItems: NavigationItem[] = [
  { label: 'Termeni și Condiții', path: '/termeni-conditii', category: 'legal' },
  { label: 'Politica de Confidențialitate', path: '/politica-confidentialitate', category: 'legal' },
  { label: 'Politica Cookie-uri', path: '/politica-cookies', category: 'legal' },
  { label: 'GDPR', path: '/gdpr', category: 'legal' },
  { label: 'Declarație Accesibilitate', path: '/accesibilitate', category: 'legal' },
  { label: 'Hartă Site', path: '/harta-site', category: 'legal' }
];

// Toate elementele de navigare împreună
export const allNavItems: NavigationItem[] = [
  ...navigationItems,
  ...additionalNavItems,
  ...legalNavItems
];

// Helper functions
export const findNavItemByPath = (path: string): NavigationItem | null => {
  const searchInItems = (items: NavigationItem[]): NavigationItem | null => {
    for (const item of items) {
      if (item.path === path) {
        return item;
      }
      if (item.children) {
        const found = searchInItems(item.children);
        if (found) return found;
      }
    }
    return null;
  };
  
  return searchInItems(allNavItems);
};

export const getNavItemsByCategory = (category: NavigationItem['category']): NavigationItem[] => {
  return allNavItems.filter(item => item.category === category);
};

File: backend/app/api/endpoints/navigation.py
"""
Endpoint-uri pentru navigarea și structura site-ului
"""
from typing import List, Dict, Any
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import select
from sqlalchemy.orm import selectinload

from ...core.database import get_async_session
from ...models.content import ContentCategory, Page, AnnouncementCategory

router = APIRouter()


@router.get("/menu")
async def get_site_menu(
    db: AsyncSession = Depends(get_async_session)
) -> Dict[str, Any]:
    """
    Obține structura de meniu pentru site conform standardelor #DigiLocal
    """
    
    # Obține categoriile principale
    categories_query = select(ContentCategory).where(
        ContentCategory.is_active == True
    ).order_by(ContentCategory.menu_order, ContentCategory.name)
    
    categories_result = await db.execute(categories_query)
    categories = categories_result.scalars().all()
    
    # Obține paginile publice pentru fiecare categorie
    pages_query = select(Page).where(
        Page.status == 'published'
    ).order_by(Page.menu_order, Page.title)
    
    pages_result = await db.execute(pages_query)
    all_pages = pages_result.scalars().all()
    
    # Construire structură de meniu
    menu_structure = {
        "main_sections": [],
        "footer_links": [
            {
                "title": "Termeni și Condiții",
                "url": "/termeni-conditii",
                "external": False
            },
            {
                "title": "Politica de Confidențialitate", 
                "url": "/politica-confidentialitate",
                "external": False
            },
            {
                "title": "GDPR",
                "url": "/gdpr", 
                "external": False
            },
            {
                "title": "Accesibilitate",
                "url": "/accesibilitate",
                "external": False
            }
        ],
        "quick_links": [
            {
                "title": "Programări Online",
                "url": "/programari-online",
                "icon": "calendar",
                "color": "#004990"
            },
            {
                "title": "Plăți Online",
                "url": "/plati-online", 
                "icon": "credit-card",
                "color": "#0079C1"
            },
            {
                "title": "Formulare Online",
                "url": "/servicii-publice/formulare",
                "icon": "file-text",
                "color": "#28a745"
            },
            {
                "title": "Contact",
                "url": "/contact",
                "icon": "phone",
                "color": "#ffc107"
            }
        ]
    }
    
    # Structura obligatorie #DigiLocal
    digilocal_structure = [
        {
            "title": "Despre Primărie",
            "url": "/despre-primarie",
            "category_id": 1,
            "submenu": [
                {"title": "Organizare", "url": "/despre-primarie/organizare"},
                {"title": "Conducere", "url": "/despre-primarie/conducere"}, 
                {"title": "Strategia de Dezvoltare Locală", "url": "/despre-primarie/strategia-dezvoltare"}
            ]
        },
        {
            "title": "Informații de Interes Public",
            "url": "/informatii-interes-public",
            "category_id": 2,
            "submenu": [
                {"title": "Buget și Execuție Bugetară", "url": "/informatii-interes-public/buget"},
                {"title": "Achiziții Publice", "url": "/informatii-interes-public/achizitii"},
                {"title": "Taxe și Impozite Locale", "url": "/informatii-interes-public/taxe-impozite"}
            ]
        },
        {
            "title": "Transparență Decizională", 
            "url": "/transparenta-decizionala",
            "category_id": 3,
            "submenu": [
                {"title": "Proiecte de Hotărâri", "url": "/transparenta-decizionala/proiecte-hotarari"},
                {"title": "Ședințe ale Consiliului Local", "url": "/transparenta-decizionala/sedinte-consiliu"}
            ]
        },
        {
            "title": "Integritate Instituțională",
            "url": "/integritate-institutionala", 
            "category_id": 4,
            "submenu": [
                {"title": "Cod Etic/Deontologic", "url": "/integritate-institutionala/cod-etic"},
                {"title": "Plan de Integritate", "url": "/integritate-institutionala/plan-integritate"}
            ]
        },
        {
            "title": "Monitorul Oficial Local",
            "url": "/monitorul-oficial-local",
            "category_id": None,
            "submenu": [
                {"title": "Statutul Unității Administrativ-Teritoriale", "url": "/mol/categoria/1"},
                {"title": "Regulamentele privind procedurile administrative", "url": "/mol/categoria/2"},
                {"title": "Hotărârile autorității deliberative", "url": "/mol/categoria/3"},
                {"title": "Dispozițiile autorității executive", "url": "/mol/categoria/4"},
                {"title": "Documente și informații financiare", "url": "/mol/categoria/5"},
                {"title": "Alte documente", "url": "/mol/categoria/6"}
            ]
        },
        {
            "title": "Servicii Publice",
            "url": "/servicii-publice",
            "category_id": 5, 
            "submenu": [
                {"title": "Formulare Online", "url": "/servicii-publice/formulare"},
                {"title": "Programări Online", "url": "/servicii-publice/programari"},
                {"title": "Plata Taxelor și Impozitelor", "url": "/servicii-publice/plata-taxelor"},
                {"title": "Urbanism și Dezvoltare", "url": "/servicii-publice/urbanism"}
            ]
        },
        {
            "title": "Comunitate",
            "url": "/comunitate",
            "category_id": 6,
            "submenu": [
                {"title": "Educație, Cultură și Sănătate", "url": "/comunitate/educatie-cultura"},
                {"title": "Mediu și Turism", "url": "/comunitate/mediu-turism"}
            ]
        },
        {
            "title": "Anunțuri",
            "url": "/anunturi", 
            "category_id": None,
            "submenu": []
        }
    ]
    
    # Adăugare pagini existente în structură
    for section in digilocal_structure:
        if section["category_id"]:
            category_pages = [p for p in all_pages if p.category_id == section["category_id"]]
            for page in category_pages[:5]:  # Limităm la 5 pagini per secțiune
                section["submenu"].append({
                    "title": page.title,
                    "url": f"/pagina/{page.slug}",
                    "page_id": page.id
                })
    
    menu_structure["main_sections"] = digilocal_structure
    
    return menu_structure


@router.get("/breadcrumbs")
async def get_breadcrumbs(
    path: str,
    db: AsyncSession = Depends(get_async_session)
) -> List[Dict[str, str]]:
    """
    Obține breadcrumb-urile pentru o cale specifică
    """
    breadcrumbs = [{"title": "Acasă", "url": "/"}]
    
    # Parsare cale simplă
    path_parts = path.strip("/").split("/")
    current_path = ""
    
    # Mapare căi către titluri
    path_titles = {
        "despre-primarie": "Despre Primărie",
        "informatii-interes-public": "Informații de Interes Public", 
        "transparenta-decizionala": "Transparență Decizională",
        "integritate-institutionala": "Integritate Instituțională",
        "monitorul-oficial-local": "Monitorul Oficial Local",
        "servicii-publice": "Servicii Publice",
        "comunitate": "Comunitate",
        "anunturi": "Anunțuri",
        "contact": "Contact",
        "programari-online": "Programări Online",
        "plati-online": "Plăți Online"
    }
    
    for i, part in enumerate(path_parts):
        if part:
            current_path += "/" + part
            title = path_titles.get(part, part.replace("-", " ").title())
            
            breadcrumbs.append({
                "title": title,
                "url": current_path
            })
    
    return breadcrumbs


@router.get("/sitemap")
async def get_sitemap(
    db: AsyncSession = Depends(get_async_session)
) -> Dict[str, Any]:
    """
    Generează harta site-ului pentru SEO și navigare
    """
    
    # Obține toate paginile publicate
    pages_query = select(Page).options(
        selectinload(Page.category)
    ).where(
        Page.status == 'published'
    ).order_by(Page.category_id, Page.menu_order)
    
    pages_result = await db.execute(pages_query)
    pages = pages_result.scalars().all()
    
    # Obține categoriile de anunțuri
    ann_categories_query = select(AnnouncementCategory).where(
        AnnouncementCategory.is_active == True
    )
    ann_categories_result = await db.execute(ann_categories_query)
    announcement_categories = ann_categories_result.scalars().all()
    
    sitemap = {
        "static_pages": [
            {"title": "Pagina Principală", "url": "/", "priority": 1.0},
            {"title": "Contact", "url": "/contact", "priority": 0.8},
            {"title": "Harta Site", "url": "/harta-site", "priority": 0.5}
        ],
        "content_pages": [],
        "services": [
            {"title": "Programări Online", "url": "/programari-online", "priority": 0.9},
            {"title": "Plăți Online", "url": "/plati-online", "priority": 0.9},
            {"title": "Formulare Online", "url": "/servicii-publice/formulare", "priority": 0.8},
            {"title": "Căutare Programări", "url": "/verificare-programare", "priority": 0.7}
        ],
        "announcements": [
            {"title": "Toate Anunțurile", "url": "/anunturi", "priority": 0.8}
        ],
        "legal_pages": [
            {"title": "Termeni și Condiții", "url": "/termeni-conditii", "priority": 0.3},
            {"title": "Politica de Confidențialitate", "url": "/politica-confidentialitate", "priority": 0.3},
            {"title": "GDPR", "url": "/gdpr", "priority": 0.3},
            {"title": "Accesibilitate", "url": "/accesibilitate", "priority": 0.3}
        ]
    }
    
    # Adăugare pagini de conținut
    for page in pages:
        sitemap["content_pages"].append({
            "title": page.title,
            "url": f"/pagina/{page.slug}",
            "category": page.category.name if page.category else None,
            "priority": 0.7,
            "last_modified": page.updated_at.isoformat() if page.updated_at else None
        })
    
    # Adăugare categorii de anunțuri
    for category in announcement_categories:
        sitemap["announcements"].append({
            "title": f"Anunțuri - {category.name}",
            "url": f"/anunturi?categoria={category.slug}",
            "priority": 0.6
        })
    
    return sitemap